pkgs/by-name/ne/nexusmods-app/package.nix

```diff
@@ -51,9 +51,12 @@ buildDotnetModule rec {
     ln --force --symbolic "${lib.getExe _7zzWithOptionalUnfreeRarSupport}" src/ArchiveManagement/NexusMods.FileExtractor/runtimes/linux-x64/native/7zz
   '';
 
+  NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR = 1; # Use nixpkgs 7zz executable
+
   makeWrapperArgs = [
     "--prefix PATH : ${lib.makeBinPath [desktop-file-utils]}"
     "--set APPIMAGE $out/bin/${meta.mainProgram}" # Make associating with nxm links work on Linux
+    "--set INSTALLATION_METHOD_PACKAGE_MANAGER 1" # Tell the app it's a distro package
```

Comment:
[Upstream docs](https://github.com/Nexus-Mods/NexusMods.App/blob/main/docs/Contributing.md#for-package-maintainers) say to set the documented vars "when publishing".

Have you checked if they mean at build time or at runtime?

(I assume wrapper args will affect the app at runtime)

Reply:
@erri120 would it be possible to clarify your publishing docs, to specify if these variables apply when building or when running the app?

Reply:
I would assume this means at runtime, since otherwise the variable would have to be set in some file which is sourced at runtime, or compiled into the program.

Reply:
> or compiled into the program.

I think that's the case. Looking at https://github.com/Nexus-Mods/NexusMods.App/commit/95782f541987b7b6f5fe5bbf09b6c5b57412a944, these are build props, not than env vars.

Reply:
[You're right](https://github.com/Nexus-Mods/NexusMods.App/commit/95782f541987b7b6f5fe5bbf09b6c5b57412a944#diff-4fb37a10b38749d0e7315711dbe7a65f3d06e337055721656a8df6bd559ebfedR12). Now to figure out how to set build properties in `buildDotnetModule`. I guess that's in [`dotnetBuildFlags`](https://github.com/NixOS/nixpkgs/blob/cd7ec2419a9a98e9119d64f5b0cbaa069e53c298/doc/languages-frameworks/dotnet.section.md#packaging-a-dotnet-application-packaging-a-dotnet-application).

Reply:
Although `INSTALLATION_METHOD_PACKAGE_MANAGER` is not in 0.5.1, that'll need the upcoming release.

- `NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR` [exists in 0.5.1](https://github.com/Nexus-Mods/NexusMods.App/blob/v0.5.1/Directory.Build.props#L17C17-L17C51)
- `INSTALLATION_METHOD_PACKAGE_MANAGER` was [named `INSTALLATION_METHOD_FLATPAK`](https://github.com/Nexus-Mods/NexusMods.App/blob/v0.5.1/Directory.Build.props#L27C17-L27C44) at the time

Reply:
> [You're right](https://github.com/Nexus-Mods/NexusMods.App/commit/95782f541987b7b6f5fe5bbf09b6c5b57412a944#diff-4fb37a10b38749d0e7315711dbe7a65f3d06e337055721656a8df6bd559ebfedR12). Now to figure out how to set build properties in `buildDotnetModule`. I guess that's in [`dotnetBuildFlags`](https://github.com/NixOS/nixpkgs/blob/cd7ec2419a9a98e9119d64f5b0cbaa069e53c298/doc/languages-frameworks/dotnet.section.md#packaging-a-dotnet-application-packaging-a-dotnet-application).

https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-build

Looks like you want `-p` or `--property` in `dotnetBuildFlags`

Reply:
WTH? That is not described in `nix-shell --packages dotnet-sdk_8 --run 'dotnet build --help'`. Ah well.

Reply:
I've gone with the relevant property for now; we can update it when upgrading.

Reply:
> when publishing

`dotnet build` is a generic build command, while `dotnet publish` is a command made specifically for publishing executables. We also use https://github.com/kuiperzone/PupNet-Deploy to create AppImages.

You should consider using `dotnet publish` instead of `dotnet build` if possible, as it allows you to do stuff like AOT compiling, single-file publishing, and stripping unnecessary parts.